hardware/detect.py

```diff
@@ -726,6 +687,86 @@ def find_element(xml, xml_spec, sys_subtype,
     return True
 
 
+def get_cpus(hw_lst, lscpu_output=None, lscpux_output=None, scaling_governor_output=None):
+    # Extracting lspcu information
+    lscpu = {}
+    if lscpu_output is None:
+        output = detect_utils.output_lines('LANG=en_US.UTF-8 lscpu')
+    else:
+        output = lscpu_output.splitlines()
+    for line in output:
+        item, value = line.split(':')
+        lscpu[item.strip(':')] = value.strip()
+
+    # Extracting lspcu -x information
+    lscpux = {}
+    if lscpux_output is None:
+        output = detect_utils.output_lines('LANG=en_US.UTF-8 lscpu -x')
+    else:
+        output = lscpux_output.splitlines()
+    for line in output:
+        item, value = line.split(':')
+        lscpux[item.strip(':')] = value.strip()
+
+    hw_lst.append(("cpu", "physical", "number", int(lscpu["Socket(s)"])))
+    for processor in range(int(lscpu["Socket(s)"])):
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'vendor', lscpu['Vendor ID']))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'product', lscpu['Model name']))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'cores', int(lscpu['Core(s) per socket'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'threads', int(lscpu['Thread(s) per core']) * int(lscpu['Core(s) per socket'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'family', int(lscpu['CPU family'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'model', int(lscpu['Model'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'stepping', int(lscpu['Stepping'])))
+        for item in ['L1d cache', 'L1i cache', 'L2 cache', 'L3 cache']:
+            hw_lst.append(('cpu', "physical_{}".format(
+                processor), item.lower(), lscpu[item]))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'min_Mhz', float(lscpu['CPU min MHz'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'max_Mhz', float(lscpu['CPU max MHz'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'current_Mhz', float(lscpu['CPU MHz'])))
+        hw_lst.append(('cpu', "physical_{}".format(
+            processor), 'flags', lscpu['Flags']))
+
+    hw_lst.append(('cpu', 'logical', 'number', int(lscpu['CPU(s)'])))
+    # Govenors could be differents on logical cpus
+    for cpu in range(int(lscpu['CPU(s)'])):
+        if scaling_governor_output is None:
+            scaling_governor = detect_utils.output_lines(
+                "cat /sys/devices/system/cpu/cpufreq/policy{}/scaling_governor".format(cpu))
+        else:
+            scaling_governor = scaling_governor_output
```

Comment:
It seems like if `scaling_governor_output` is provided by the caller, it's the same for all CPUs? Was it your intention?

Reply:
Could you answer this question please @ErwanAliasr1?

Reply:
Sorry I did answered but wasn't saved...
It's a way to trick the input during tests, not that beautiful.

Reply:
Why not mock `output_lines` and drop these input parameters completely?

Reply:
I'll test that

Reply:
This is done.